### PR TITLE
Feature/multi file generation

### PR DESCRIPTION
**Pull request by kreuzerk**

### CODE REVIEW

**Comment by kreuzerk** [comment]
@Palapapaa FYI. I will document this further. Will write a blog post about this feature - will let you know when I am done.

File: src/bin/svg-to-ts.ts [hunk 1]
```diff
@@ -21,7 +24,8 @@ const DEFAULTS = {
   outputDirectory: './dist',
   prefix: 'myIcon',
   sourceFilesRegex: ['*.svg'],
-  typeName: 'myIcons'
+  typeName: 'myIcons',
+  optimizeForLayzLoading: false
```
Comment (Palapapaa) [inline src/bin/svg-to-ts.ts]: There seem's to be a typo here

Comment (kreuzerk) [reply src/bin/svg-to-ts.ts]: What typo, I don't see it ;)

Comment (Palapapaa) [reply src/bin/svg-to-ts.ts]: `optimizeForLayzLoading` should be `optimizeForLazyLoading` right ?

Comment (kreuzerk) [reply src/bin/svg-to-ts.ts]: Ah ok, yes y and z is wrong. Sorry, had a hard time seeing a difference 😅will correct it

**Comment by Palapapaa** [comment]
All this changes look promising =)

I understand the previous refactoring pull request now =)

I'll take a look at this.

Nice job !

**Comment by kreuzerk** [comment]
Yeah, it's based on some ideas I had in mind for a long time. I finally got time to dive a little bit deeper into this topic. But I still need to document this. The basic idea is that the whole library becomes even more tree shakable. Especially when it is used in combination with lazy loading. 

For now, I would say, this feature is still experimental. Therefore no official docs. I will try to use this in a real-world project and then document it accordingly. There are some additional manual steps you need to perform. The current implementation is not affected by this change it therefore continues to work as usual.

**Pull request merged**